Automattic/wp-calypso
# Import: improve the error message when user tries to upload a wpress file

#### Proposed Changes

* To improve the user experience, make the error message more helpful when a user tries to upload a `.wpress` file using `/import`
* Introduced a new action type - `IMPORTS_PRE_UPLOAD_FAILED` and defined its corresponding action creator and reducing function to handle the cross-components communication between `uploading-pane` and `error-pane`. A `IMPORTS_PRE_UPLOAD_FAILED` action will be dispatched by the `uploading-pane` component when it detects the uploading file is a `.wpress` file in the `setupUpload()` function call. The `error-pane` will display the improved error message when it is notified that the error status has changed.

#### Testing Instructions

* Go to http://calypso.localhost:3000/import/{SITE_SLUG}
* Select import content from `WordPress`

<img width="888" alt="Screen Shot 2022-08-22 at 3 56 58 PM" src="https://user-images.githubusercontent.com/1024985/185869605-9b75fee2-9f09-4968-9d73-8c84cac01dc7.png">

*  Click `upload it to import content`
*  Upload a `.wpress` file (Here is a [file](https://drive.google.com/file/d/1wqBy7iN5iyxJd0nRqy1OnIBj8Uwcw6iP/view?usp=sharing) for testing)
*  You should see this error message appear after the uploading is completed

<img width="848" alt="Screen Shot 2022-08-24 at 8 32 51 PM" src="https://user-images.githubusercontent.com/1024985/186424218-8b24962f-438c-4198-9cd3-a9e0890371c7.png">

* Click the `install the plugin` link should head to `/plugins/all-in-one-wp-migration`

<img width="834" alt="Screen Shot 2022-08-22 at 4 04 34 PM" src="https://user-images.githubusercontent.com/1024985/185870937-dfbbd5f7-ba49-44e3-9b26-72988def3265.png">

* Click the `try out Everything import` link should head to `http://calypso.localhost:3000/setup/import?siteSlug={SITE_SLUG}`

<img width="1487" alt="Screen Shot 2022-08-22 at 3 14 05 PM" src="https://user-images.githubusercontent.com/1024985/185871213-e3db19c4-979e-4f23-a8e5-1c5420ffbe6e.png">


#### Pre-merge Checklist






Related to #
